### ISSUE TITLE
Partial user id causes ArrayIndexOutOfBoundsException (Error in OM layer)

### ISSUE THREAD

**Issue by cobratbq** [migzai/irc-api#24]
A user of the Jitsi IRC protocol support I'm working on encountered the runtime exception below. As I understand this was on a private IRC server. I haven't asked for the exact IRC server software, which is probably useful/relevant here.

I checked out the code and it looks like the code fails assuming that the `<nick>!<user>@<host>` is used, even though this format is not fully present in the server message. I have not seen the raw message though, so I can't say for sure.

This issue happens when the user is joining (or attempting to join) a channel. He will (at least some times) actually join, however it appears that since the flow is interrupted, Jitsi never actually gets a callback from irc-api. So I can't be any more specific than this.

I don't have an answer readily available on whether this is bad behaviour on the part of the server or whether this is a valid case not currently supported by irc-api.

~~~
 15:04:54.343 SEVERE: [289] com.ircclouds.irc.api.AbstractMessageReader.readMessage() Error from the OM layer
com.ircclouds.irc.api.om.IRCOMException: java.lang.ArrayIndexOutOfBoundsException: 1
    at com.ircclouds.irc.api.om.AbstractMessageFactory.build(AbstractMessageFactory.java:138)
    at com.ircclouds.irc.api.AbstractMessageReader.readMessage(AbstractMessageReader.java:64)
    at com.ircclouds.irc.api.AbstractApiDaemon.run(AbstractApiDaemon.java:32)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
    at com.ircclouds.irc.api.utils.ParseUtils.getUser(ParseUtils.java:32)
    at com.ircclouds.irc.api.om.AbstractPrivateMessageBuilder.build(AbstractPrivateMessageBuilder.java:20)
    at com.ircclouds.irc.api.om.AbstractMessageFactory.build(AbstractMessageFactory.java:90)
    ... 2 more
~~~

**Comment by qirtaiba** [comment]
(1) ircd-hybrid 7.2 on Debian (http://www.ircd-hybrid.org/)
(2) Whatever slack.com uses as their irc gateway (https://slack.com/)

**Comment by cobratbq** [comment]
I fixed this in 0c200c644a7c85a024101ca95e789d3fb0064862
